src/app/CommandHandlerImpl.cpp

```diff
@@ -65,7 +65,10 @@ CHIP_ERROR CommandHandlerImpl::AllocateBuffer()
     {
         mCommandMessageWriter.Reset();
 
-        System::PacketBufferHandle commandPacket = System::PacketBufferHandle::New(chip::app::kMaxSecureSduLengthBytes);
+        System::PacketBufferHandle commandPacket = nullptr;
+        size_t commandBufferMaxSize              = mpResponder->GetCommandResponseMaxBufferSize();
+        commandPacket                            = System::PacketBufferHandle::New(commandBufferMaxSize);
```

Comment:
```suggestion
        const size_t commandBufferMaxSize  = mpResponder->GetCommandResponseMaxBufferSize();
        auto commandPacket.                           = System::PacketBufferHandle::New(commandBufferMaxSize);
```

Why the existence on declaring things before they are needed?

Reply:
Fixed.